helper/resource/testing.go

```diff
@@ -1612,6 +1921,88 @@ func testCheckResourceAttrPair(isFirst *terraform.InstanceState, nameFirst strin
 }
 
 // TestCheckOutput checks an output in the Terraform configuration
+//
+// Deprecated: State checks using [TestStep.Check] have been superseded by [TestStep.ConfigStateChecks].
+// Use the built-in [statecheck.ExpectKnownOutputValue], and
+// [statecheck.ExpectKnownOutputValueAtPath] state checks instead.
+// TestCheckOutput function will be removed in the next major version.
+//
+// The following is an example of using [statecheck.ExpectKnownOutputValue]
+// to replicate the behaviour of TestCheckOutput.
+//
+//	package example_test
+//
+//	import (
+//		"testing"
+//
+//		"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+//		"github.com/hashicorp/terraform-plugin-testing/knownvalue"
+//		"github.com/hashicorp/terraform-plugin-testing/statecheck"
+//	)
+//
+//	func TestExpectKnownOutputValue_CheckState_Bool(t *testing.T) {
+//		t.Parallel()
+//
+//		resource.Test(t, resource.TestCase{
+//			// Provider definition omitted.
+//			Steps: []resource.TestStep{
+//				{
+//					// Example resource containing a computed boolean attribute named "computed_attribute"
+//					Config: `resource "test_resource" "one" {}
+//
+//			        output computed_output {
+//			          value = test_resource.one.computed_attribute
+//			        }`,
```

Comment:
It might be best to leave an additional comment here that this testing practice is generally not necessary (instead check the resource attribute directly) and only shown as an example, or choose an example like provider function testing which doesn't need a resource. 👍